File: ShearNDipityLib/src/Models/AppointmentType.java
package Models;

/**
 * Created with IntelliJ IDEA.
 * User: mindikingsun
 * Date: 25/08/13
 * Time: 9:15 PM
 * <p/>
 * Each appointment will require an appointment type.  The type will dictate the amount of time
 * required for it to fit into the scheduler.
 */

public class AppointmentType {

    private int typeId;
    private String description;
    private String duration;
    private int isAllDay;

    public AppointmentType() {
    }

    public AppointmentType(int appTypeId, String appTypeDescription, String appTypeDuration, int appTypeAllDay) {
        this.typeId = appTypeId;
        this.description = appTypeDescription;
        this.duration = appTypeDuration;
        this.isAllDay = appTypeAllDay;
    }

    public int getTypeId() {
        return typeId;
    }

    public void setTypeId(int typeId) {
        this.typeId = typeId;
    }

    public String getDescription() {
        System.out.println(description);
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getDuration() {
        return duration;
    }

    public void setDuration(String duration) {
        this.duration = duration;
    }

    public boolean getIsAllDay() {
        return isAllDay == 1;
    }

    public void setIsAllDay(boolean isAllDay) {
        this.isAllDay = isAllDay ? 1 : 0;
    }
}

File: Server/server/models/AppointmentType.js
var Ring = require('ring');
var Utilities = require('../utilities/Utilities.js');

var AppointmentType = Ring.create({
    _typeId: 0,
    _description: '',
    _duration: '00:00',
    _isAllDay: 0,
    _isActive: 1,

    init: function () {
    },

    setId: function (value) {
        if (Utilities.isIntegerAboveZero(value)) {
            this._typeId = 0;
        }
    },
    getId: function () {
        return this._typeId;
    },

    setDuration: function (value) {
        if (Utilities.isTimeStamp(value)) {
            this._duration = value;
        }
    },
    getDuration: function () {
        return this._duration;
    },

    setIsAllDay: function (value) {
        this._isAllDay = value ? 1 : 0;
    },

    getIsAllDay: function () {
        return this._isAllDay;
    },
    setDescription: function (value) {
        if (Utilities.isStringAndNotEmpty(value)) {
            this._description = value;
        }
    },

    getDescription: function () {
        return this._description;
    },

    getIsActive: function () {
        return this._isActive;
    },

    setIsActive: function (value) {
        this._isActive = value ? 1 : 0;
    },

    toJSON: function () {
        return {
            "typeId": this._typeId,
            "description": this._description,
            "duration": this._duration,
            "isAllDay": this._isAllDay,
            "isActive": this._isActive
        }
    }
});

AppointmentType.fromJSON = function (data) {
    var type = new AppointmentType();

    type.setId(data.typeId);
    console.log(data);
    type.setDuration(data.duration);
    type.setDescription(data.description);
    type.setIsAllDay(data.isAllDay);
    return type;
};

module.exports = AppointmentType;